ast/ast.go

```diff
@@ -25,9 +27,35 @@ type Expression interface {
 	expressionNode()
 }
 
+// PositionInterface all nodes have this interface
+type PositionInterface interface {
+	SetPosition(int, *lexer.Lexer)
+	GetLine() (int, int, string)
+}
+
+// Position defines the offset into the file where this node was seen
+type Position struct {
+	Position int          // records our Position in the file
+	Lexer    *lexer.Lexer // pointer to our Lexer instance for error location in Eval()
+}
+
+// SetPosition sets the Position and Lexer for this node
+func (p *Position) SetPosition(pos int, lex *lexer.Lexer) {
+	p.Position = pos
+	p.Lexer = lex
+}
+
+// GetLine returns (lineNum, column, thisLine) from this node's Position
+func (p *Position) GetLine() (int, int, string) {
+	pos := p.Position
```

Comment:
this variable isn't neede, you can pass `p.Position` to `GetLinePos(...)` directly